### PR TITLE
feat: add OS template property

### PR DESCRIPTION
**Pull request by gitolicious**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

Your code looked really clean and easy to follow so I gave #683 a shot. The following example results in a nice [Debian rubine red](https://wiki.debian.org/DebianLogo#The_red_color) OS logo:

```jsonc
        {
          "type": "os",
          "style": "plain",
          "foreground": "#ffffff",
          "background_templates": [
            "{{if contains \"debian\" .OS}}#CE0056{{end}}",
            "#333333"
          ]
        },
```

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by gitolicious** [comment]
Potential extensions I could think of are
* a `.WSL` template property to define different colors for WSL and native consoles
* a `"color_background": true` configuration property which automatically sets pre-defined background colors for each OS

File: src/segment_os.go [hunk 1]
```diff
@@ -76,6 +77,7 @@ func (n *osInfo) string() string {
 	case "linux":
 		wsl := n.env.getenv("WSL_DISTRO_NAME")
 		p := n.env.getPlatform()
+		n.OS = p
```
Comment (JanDeDobbeleer) [inline src/segment_os.go]: I would opt to set this regardless of the platform:

```golang
func (n *osInfo) string() string {
	goos := n.env.getRuntimeGOOS()
	switch goos {
	case windowsPlatform:
		n.OS = windowsPlatform
		return n.props.getString(Windows, "\uE62A")
	case darwinPlatform:
		n.OS = darwinPlatform
		return n.props.getString(MacOS, "\uF179")
	case linuxPlatform:
		wsl := n.env.getenv("WSL_DISTRO_NAME")
		p := n.env.getPlatform()
		if len(wsl) == 0 {
			n.OS = p
			return n.getDistroName(p, "")
		}
		n.OS = wsl
		return fmt.Sprintf("%s%s%s",
			n.props.getString(WSL, "WSL"),
			n.props.getString(WSLSeparator, " - "),
			n.getDistroName(p, wsl))
	default:
		n.OS = goos
		return goos
	}
}
```

This also add the two platforms to `environment.go`:

```golang
	darwinPlatform  = "darwin"
	linuxPlatform   = "linux"
```

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Comment by JanDeDobbeleer** [comment]
> a "color_background": true configuration property which automatically sets pre-defined background colors for each OS

@lnu that's solvable by using templates imho. Our go-to should be that, if the template can handle it, we don't extend the base properties. Ideally we can move 100% to templates for every segment one day.

File: src/segment_os.go (hunk 1 above)
Comment (gitolicious) [reply src/segment_os.go]: Good point, I amended the PR accordingly.

**Comment by gitolicious** [comment]
> @lnu that's solvable by using templates imho. Our go-to should be that, if the template can handle it, we don't extend the base properties. Ideally we can move 100% to templates for every segment one day.

A predefined  setting would save the users looking up the appropriate colors for the 20+ distros, though finding nice defaults would be quite some effort even for the repo maintainers.

**Comment by JanDeDobbeleer** [comment]
@gitolicious true, I'd like to meet the user who has them all set up though...

**Review by JanDeDobbeleer — APPROVED** [review]

**Comment by gitolicious** [comment]
Thx for the quick merge and kudos for your automation and toolchain setup on Github! Never saw <5 mins from merging a PR to having the new version installed on my system via Scoop. 🏎

**Comment by lnu** [comment]
@JanDeDobbeleer @gitolicious It's a trade-off between simplicity/ease of use/predefined settings and more advanced/customizable settings with templates. The first way means more work/test and magic, the second one if well documented(with samples in the doc) can be very powerful.
I'd like both ways but as most of omp users are already "power users", the second strategy can be more efficient on the long term. Both approaches makes sense, now it's up to the "lead" to make a decision :grin:

**Comment by gitolicious** [comment]
I looked up some colors codes for the more popular choices.

`macos`: `#000000` / `#a3aaae`
image
`windows`: `#ffffff` / `#1675c7` (PowerShell logo colors)
image
`alpine`: `#ffffff` / `#00567e`
image
`debian`: `#ffffff` / `#ce0056`
image
`raspbian`: `#ffffff` / `#c51a4a`
image
`fedora`: `#ffffff` / `#294172`
image
`gentoo`: `#ffffff` / `#54487a`
image
`opensuse`: `#ffffff` / `#73ba25`
image
`ubuntu`: `#ffffff` / `#e95420`
image

**Pull request merged**